File: tRNA-folding/makemap.py
import pandas as pd
import numpy as np

contRes=pd.read_csv("reslist", delim_whitespace=True,header=None)



resmap=pd.read_csv("resmap",header=None,squeeze=True)

conts=pd.read_csv("arg.cont",header=None,delim_whitespace=True)

##conts.reindex(resmap)

##conts['4']=contRes.iloc[resmap.iloc[conts[0]-1]+1]

conts['4']= resmap


#conts['5']= contRes.iloc[conts['4']-1,:]

conts['5']=contRes.loc[conts['4']-1,:].loc[:,0]

print(conts.head())

#print(contRes.loc[conts['4']-1,:])

File: tRNA-folding/makemap.pl
#use warnings;

open(reslist,"reslist");
my %resContlist;
my $counter=0;
while(<reslist>){
    $counter++;
    my $line=$_;
    chomp($line);
    $line =~  s/^\s+|\s+$//g;
    $line =~ s/\s+/ /g;
    $resContlist{$counter}=$line;
}
close(reslist);


open(resmap,"resmap");
my %resContmap;
$counter=0;
while(<resmap>){
    $counter++;
    my $line=$_;
    chomp($line);
    $line =~ s/^\s+|\s+$//g;
    $resContmap{$counter}=$line;
}
close(resmap);


open(pairlist,"arg.cont");

open(pairsout,">contres");

$counter=0;
while(<pairlist>){
    $counter++;
    my $line=$_;
    chomp($line);
    $line =~ s/^\s+|\s+$//g;
    $line =~ s/\s+/ /g;
    @A=split(/ /,$line);
    @B=split(/ /,$resContlist{$resContmap{$counter}});
    print pairsout "$A[0] $A[1] $B[0] $B[1]\n";
}

close(pairlist);
close(pairsout);
